layer/nrlf/core/validators.py

```diff
@@ -316,3 +317,52 @@ def _validate_ssp_asid(self, model: DocumentReference):
                 field=f"context.related[{idx}].identifier.value",
             )
             return
+
+    def _validate_category(self, model: DocumentReference):
+        """
+        Validate the category field contains an appropriate coding system, code and display.
+        """
+
+        if len(model.category) > 1:
+            logger.log(
+                LogReference.VALIDATOR001, step="category", reason="category_too_long"
+            )
+            self.result.add_error(
+                issue_code="invalid",
+                error_code="INVALID_RESOURCE",
+                diagnostics=f"Invalid category length: {len(model.category)} Category must only contain a single value",
+                field=f"category",
+            )
+            return
+
+        logger.log(LogReference.VALIDATOR001, step="category")
+
+        logger.debug("Validating category")
+
+        for index, coding in enumerate(model.category[0].coding):
+            if coding.system != "http://snomed.info/sct":
+                self.result.add_error(
+                    issue_code="value",
+                    error_code="INVALID_RESOURCE",
+                    diagnostics=f"Invalid category system: {coding.system} Category system must be 'http://snomed.info/sct'",
+                    field=f"category[0].coding[{index}].system",
+                )
+                continue
+
+            if coding.code not in CATEGORIES.keys():
+                self.result.add_error(
+                    issue_code="value",
+                    error_code="INVALID_RESOURCE",
+                    diagnostics=f"Invalid category code: {coding.code} Category code must be '734163000' or '1102421000000108'.",
```

Comment:
This is good but as we have at least four new categories planned in the near future, instead of needing to update this all the time we could just tell them what the Value Set is (because it will be dynamically updated). Something like, "Category must be a member of the England-NRLRecordCategory value set (https://fhir.nhs.uk/England/CodeSystem/England-NRLRecordCategory)"